esmvalcore/local.py

```diff
@@ -422,110 +397,82 @@ def _select_drs(input_type, drs, project):
+            if nonexistent and (key, nonexistent) not in _ROOTPATH_WARNED:
                 logger.warning(
                     "'%s' rootpaths '%s' set in config-user.yml do not exist",
                     key, ', '.join(str(p) for p in nonexistent))
-                ROOTPATH_WARNED.add((key, nonexistent))
+                _ROOTPATH_WARNED.add((key, nonexistent))
             return rootpath[key]
     raise KeyError('default rootpath must be specified in config-user file')
 
 
-def _find_input_dirs(variable, rootpath, drs):
-    """Return a the full paths to input directories."""
+def _get_globs(variable):
+    """Compose the globs that will be used to look for files."""
     project = variable['project']
 
-    root = get_rootpath(rootpath, project)
-    path_template = _select_drs('input_dir', drs, project)
+    rootpaths = _get_rootpath(project)
 
-    dirnames = []
-    for dirname_template in _replace_tags(path_template, variable):
-        for base_path in root:
-            dirname = os.path.join(base_path, dirname_template)
-            dirname = _resolve_latestversion(dirname)
-            if dirname is None:
-                continue
-            matches = glob.glob(dirname)
-            matches = [match for match in matches if os.path.isdir(match)]
-            if matches:
-                for match in matches:
-                    dirnames.append(match)
-            else:
-                logger.debug("Skipping non-existent %s", dirname)
+    dirname_template = _select_drs('input_dir', project)
+    dirname_globs = _replace_tags(dirname_template, variable)
 
-    return dirnames
+    filename_template = _select_drs('input_file', project)
+    filename_globs = _replace_tags(filename_template, variable)
 
+    globs = sorted(r / d / f for r in rootpaths for d in dirname_globs
+                   for f in filename_globs)
+    return globs
 
-def _get_filenames_glob(variable, drs):
-    """Return patterns that can be used to look for input files."""
-    path_template = _select_drs('input_file', drs, variable['project'])
-    filenames_glob = _replace_tags(path_template, variable)
-    return filenames_glob
 
+def _get_input_filelist(variable):
+    """Return the full path to input files."""
+    variable = dict(variable)
+    if 'original_short_name' in variable:
+        variable['short_name'] = variable['original_short_name']
 
-def _find_input_files(variable, rootpath, drs):
-    """Find available input files.
+    globs = _get_globs(variable)
+    logger.debug("Looking for files matching %s", globs)
 
-    Return the files, the directory in which they are located in, and
-    the file name.
-    """
-    short_name = variable['short_name']
-    variable['short_name'] = variable['original_short_name']
-    input_dirs = _find_input_dirs(variable, rootpath, drs)
-    filenames_glob = _get_filenames_glob(variable, drs)
-    files = find_files(input_dirs, filenames_glob)
-    variable['short_name'] = short_name
```

Comment:
It's used when you're mixing CMIP5 and CMIP6. Sometimes the variable names are different and it was felt that it would be good to homogenize that. This was implemented in https://github.com/ESMValGroup/ESMValCore/pull/595 and there is a bit of documentation [here](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/overview.html#variable-and-dataset-definitions).

Reply:
It's handled differently in the new code because I wanted data finding to work if you provided e.g. `short_name='tas'` to the function `find_files` (without the modification you would have to call `find_files` with `original_short_name='tas'`).

The reason for not resetting is that the `variable` dict is copied on line 438, so we modify a copy instead of the original and therefore it is not necessary to restore the original.